src/core/cells.cpp

```diff
@@ -68,6 +73,122 @@ double max_range = 0.0;
 unsigned resort_particles = Cells::RESORT_NONE;
 int rebuild_verletlist = 1;
 
+/**
+ * @brief Search the neighborhood of x for particles
+ *        within a sphere around a point
+ *
+ * Search is limited up to the maxrange of the cell system,
+ * so it only find particles close than max range.
+ *
+ * @param x Point at which so search
+ * @param r Search radius.
+ * @return  List of particles nearby.
+ */
+std::vector<int> local_find_nearby_particles(const Vector3d &x, double r) {
+  using boost::accumulate;
+  using Utils::Span;
+
+  auto search_cell = cell_structure.position_to_cell(x);
+  auto const r2 = r * r;
+
+  auto nearby = [&](Cell *c, std::vector<int> init) {
+    return accumulate(Span<const Particle>(c->part, c->n), init,
+                      [&](std::vector<int> parts, const Particle &p) {
+                        if (r2 <= get_mi_vector(x, p.r.p).norm2()) {
+                          parts.push_back(p.identity());
+                        }
+
+                        return parts;
+                      });
+  };
+
+  return accumulate(
+      search_cell->neighbors().all(), nearby(search_cell, {}),
+      [&](std::vector<int> parts, Cell *n) { return nearby(n, parts); });
+}
+
+void cells_find_nearby_particles_slave(int pnode, int) {
+  if (comm_cart.rank() == pnode) {
+    std::pair<Vector3d, double> params;
+    comm_cart.recv(0, 52, params);
```

Comment:
Could we introduce a macro here instead of using a plain magic number 52? It would be more readable.

Reply:
And what would you call the 'macro'? `NUMBER`?

Reply:
Hehe, sorry i just saw that the magic number is the tag which is used for the MPI communication. If we would introduce a name here it would probably be function specific... So probably just forget the comment